### ISSUE TITLE
Fix L001 adds too many newlines and duplicates from clause

### ISSUE THREAD

**Issue by joellabes** [sqlfluff/sqlfluff#647]
When using the dbt compiler with sqlfluff 0.4.0a3 (Python 3.8.3 Windows), this model:

```sql
with src as (
    select * from {{ source('classes', 'classsubjectlinks') }}
),

renamed as (
    select
        classid as class_id, 
        baselanguage as base_language_id,
        targetsubject as target_subject_id,
        namematchscore as name_match_score,
        usagescore as usage_score,
        coalesce(manualscore, 0) as manual_score,
        coalesce(manualscore, (namematchscore + usagescore) / 2) as overall_score
    from src
),

ranked as (
    select
        *,
        row_number() over (
            partition by class_id
            order by
                overall_score desc,
                (name_match_score + usage_score) / 2 desc,
                target_subject_id asc
        ) as ranked_match_quality
    from renamed
)

select *
from ranked
order by class_id
```

returns this warning
```
L:   7 | P:  29 | L001 | Unnecessary trailing whitespace.
```

Running `sqlfluff fix` causes this to be generated:

```sql
with src as (

    select * from {{ source('classes', 'classsubjectlinks') }}

),



renamed as (
select
        classid as class_id,
        baselanguage as base_language_id,
        targetsubject as target_subject_id,
        namematchscore as name_match_score,
        usagescore as usage_score,
        coalesce(manualscore, 0) as manual_score,
        coalesce(manualscore, (namematchscore + usagescore) / 2) as overall_score
    from src
    from src

),



ranked as (

    select

        *,

        row_number() over (

            partition by class_id

            order by

                overall_score desc,

                (name_match_score + usage_score) / 2 desc,

                target_subject_id asc

        ) as ranked_match_quality

    from renamed

)



select *

from ranked

order by class_id
```

The newlines could be a matter of opinion _I guess_, but it looks like a bug to me. It's certainly not what I'd expect from an automated fix to a problem that was decscribed as "an unnecessary space character at the end of line 7". 

Note also the duplicate `from src` lines.

**Comment by joellabes** [comment]
When using the standard Jinja templater, this fix applies correctly - the only thing that changes is the redundant space at the end of line 7 is removed.

**Comment by reubster** [comment]
I think I'm experiencing this. I'm using vscode and it works correctly when I switch the line ending on the file to LF, but messes up when I set the file line ending to CRLF. 

Start: 

```
{{
    config(
        materialized = 'view')
}}

select
    *
from
    (values(5, 'ReubMock') as person(id, name))
```

Fix all with file set to CRLF 

```
{{

    config(

        materialized = 'view')

}select
  *
from
  (values(5, 'ReubMock') as PERSON(ID, NAME)), name))
```

    
Fix all with file set to LF 

```
{{
    config(
        materialized = 'view')
}}

select
  *
from
  (values(5, 'ReubMock') as PERSON(ID, NAME))
```

**Comment by barrywhart** [comment]
I tested this today in version 0.6.9 and got the following output.
```
WITH src AS (
    SELECT * FROM {{ source('classes', 'classsubjectlinks') }}
),

renamed AS (
    SELECT
        classid AS class_id,
        baselanguage AS base_language_id,
        targetsubject AS target_subject_id,
        namematchscore AS name_match_score,
        usagescore AS usage_score,
        coalesce(manualscore, 0) AS manual_score,
        coalesce(
            manualscore, (namematchscore + usagescore) / 2
        ) AS overall_score
    FROM src
),

ranked AS (
    SELECT
        *,
        row_number() OVER (
            PARTITION BY class_id
            ORDER BY
                overall_score DESC,
                (name_match_score + usage_score) / 2 DESC,
                target_subject_id ASC
        ) AS ranked_match_quality
    FROM renamed
)

SELECT *
FROM ranked
ORDER BY class_id
```

Looks like this is fixed. Please comment or create a new issue if you are still seeing problems.
